gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/UserServiceImpl.java

```diff
@@ -1287,7 +1287,17 @@ public Page<UserEntity> search(ExecutionContext executionContext, String query,
         SearchResult results = searchEngineService.search(executionContext, userQuery);
 
         if (results.hasResults()) {
-            List<UserEntity> users = new ArrayList<>((findByIds(executionContext, results.getDocuments())));
+            Set<UserEntity> fetched = findByIds(executionContext, results.getDocuments());
```

Comment:
```
     if (results.hasResults()) {
    Set<UserEntity> fetched = findByIds(executionContext, results.getDocuments());
    Map<String, UserEntity> byId = fetched.stream()
        .collect(Collectors.toMap(UserEntity::getId, u -> u));

    List<UserEntity> users = new ArrayList<>(results.getDocuments().size());
    Set<String> seen = new HashSet<>();

    for (String id : results.getDocuments()) {
        if (seen.add(id)) {
            UserEntity u = byId.get(id);
            if (u != null) {
                users.add(u);
            }
        }
    }

    populateUserFlags(executionContext.getOrganizationId(), users);
}     
```

Would it work with something like this ? I find it a little bit more readable like this.